File: php/features.php
<?php
declare(strict_types=1);

// AlanPerlisQuotes SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AlanPerlisQuotesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AlanPerlisQuotesBaseFeature();
            case "test":
                return new AlanPerlisQuotesTestFeature();
            default:
                return new AlanPerlisQuotesBaseFeature();
        }
    }
}

File: py/features.py
# AlanPerlisQuotes SDK feature factory

from feature.base_feature import AlanPerlisQuotesBaseFeature
from feature.test_feature import AlanPerlisQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AlanPerlisQuotesBaseFeature(),
        "test": lambda: AlanPerlisQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
